include/flang/lower/OpBuilder.h

```diff
@@ -55,6 +61,107 @@ class SymMap {
   }
 };
 
+/// Helper class that can be inherited from in order to
+/// facilitate mlir::OpBuilder usage.
+class OpBuilderHandler {
+public:
+  OpBuilderHandler(mlir::OpBuilder &b, mlir::Location l) : builder{b}, loc{l} {}
+  template <typename T, typename... Args>
+  auto create(Args... args) {
+    return builder.create<T>(loc, std::forward<Args>(args)...);
+  }
+  mlir::Value getIntegerConstant(mlir::Type integerType, std::int64_t);
```

Comment:
We really should start getting in the habit of adding doxygen to the public interfaces of things.  It'll make merging to the LLVM project that much easier.

Reply:
Reworked a bit the interface comments to add doxygen.